numbat/src/bytecode_interpreter.rs

```diff
@@ -181,18 +181,29 @@ impl BytecodeInterpreter {
                 self.vm.add_op1(Op::LoadConstant, index);
             }
             Expression::String(_, string_parts) => {
+                let mut len: u16 = 0; // TODO: this can overflow
                 for part in string_parts {
+                    len += 1;
                     match part {
                         StringPart::Fixed(s) => {
                             let index = self.vm.add_constant(Constant::String(s.clone()));
                             self.vm.add_op1(Op::LoadConstant, index)
                         }
-                        StringPart::Interpolation(_, expr) => {
+                        StringPart::Interpolation {
+                            expr,
+                            span: _,
+                            specifiers,
+                        } => {
+                            len += 1;
```

Comment:
I don't understand. If we add a `Constant::FormatSpecifier` unconditionally for every interpolation (which is good), why complicate the logic by incrementing "len" by 1 here *and* in the VM?

Reply:
No, you're totally right, I got rid of that.